linkcheck/parser/sitemap.py

```diff
@@ -29,22 +29,26 @@ def __init__(self, tag):
         self.tag = tag
         self.parser = ParserCreate()
         self.parser.buffer_text = True
-        self.parser.returns_unicode = True
+        try:
+            self.parser.returns_unicode = True
+        except AttributeError:
+            pass  # Python 3
         self.parser.StartElementHandler = self.start_element
         self.parser.EndElementHandler = self.end_element
         self.parser.CharacterDataHandler = self.char_data
 
     def parse(self, url_data):
         """Parse XML URL data."""
         self.url_data = url_data
-        self.loc = False
+        self.in_tag = False
         self.url = u""
-        data = url_data.get_raw_content()
+        data = url_data.get_content()
         isfinal = True
         try:
             self.parser.Parse(data, isfinal)
```

Comment:
This explodes on Python 2 as soon as the sitemap.xml contains any non-ASCII characters.  Try this:

```diff
diff --git a/tests/checker/data/sitemap.xml b/tests/checker/data/sitemap.xml
index 30e23048..b0ae9a20 100644
--- a/tests/checker/data/sitemap.xml
+++ b/tests/checker/data/sitemap.xml
@@ -6,4 +6,10 @@
       <changefreq>monthly</changefreq>
       <priority>0.8</priority>
    </url>
+   <url>
+      <loc>http://www.example.com/?ascii=nø</loc>
+      <lastmod>2005-01-01</lastmod>
+      <changefreq>monthly</changefreq>
+      <priority>0.8</priority>
+   </url>
 </urlset> 
```

Reply:
sounds like we should add that to the commit and fix :)

Reply:
I should have said I thought get_raw_content() was failing on Python 3 and that was why I changed it. However, it seems to work fine so I have restored it.

Thanks for explaining the problem with non-ASCII characters. I've added the suggested test. Passing here for Python 2 and also on #249 which I have identically updated.